### ISSUE TITLE
Listener only firing the first time a file changes

### ISSUE THREAD

**Issue by jlevine22** [bevry/watchr#75]
I have a script that is reading data from YAML files. I'm using watchr to detect changes in these files and reload them accordingly. However, the listener function is only being fired the first time changes are detected.

The code I'm using looks like this:
```
var rootPath = '/my/root/path';
var watchr = require('watchr');
watchr.watch({
	path: rootPath,
	listener: function (type, path) {
		console.log('Changes detected.. doing stuff');
		// Do stuff with type/path
	}
});
```
The first time I change a file the console outputs the message above. If I change it back, or change it to something else and save, the listener does not fire.

Am I missing something about the way watchr works?

**Comment by jlevine22** [comment]
It looks like this is because of how [some text editors save files](https://github.com/joyent/node/issues/3640#issuecomment-6806347). The original file at the watched path is moved/deleted so the old watcher stops firing events. I've worked around this by creating a new watcher on the original path. I determine this by comparing the birthtime of the new stat to the previous one. I'm not sure if this is really the best way to handle this situation though. Here's what I did: https://github.com/jlevine22/watchr/commit/4fe34dcf0ca3fdc8ea40c3cda267da414aef9400

**Comment by dionjwa** [comment]
I also encountered this problem. Part of the problem is that the time between deletion+creation is not known. What I had to do is poll if a file exists at the original path for a minimum time to check this use case, and then do what jlevine22  above does in re-watching the file.

**Comment by balupton** [comment]
v2.6.0 should fix this, however v3.0.0 is also released that should fix this 👍  Happy to re-open if not

Sorry for the late response on this, 2014 and 2015 were hard years for me!
